scripts/findUnused.ts

```diff
@@ -1,32 +1,140 @@
 import { relative } from "node:path"
-import { BindingNamedNode, Project } from "ts-morph"
+import { fromHere } from "@re-/node"
+import { BindingNamedNode, Node, Project, ts } from "ts-morph"
 
 const project = new Project({
-    tsConfigFilePath: "tsconfig.references.json"
+    tsConfigFilePath: fromHere("..", "tsconfig.references.json")
 })
 
 const unused: Record<string, string[]> = {}
+const snippetFileRegex = /snippets\/\w+/
+const exportAllRegex = /export \*/
+const exportAllRenamedRegex = /\* as /
+const ignoreUnusedComment = "@ignore-unused"
+const apiExports: ApiExports[] = []
+const sourceFiles = project.getSourceFiles()
+type DeclarationInfo = {
+    filePath: string
+    sourceFileText: string
+    declarationNode: Node<ts.Node>
+    referencesLength: number
+    declarationName: string
+}
+type ApiExports = {
+    path: string
+    exportedDeclarations: string[]
+}
+const checkForIgnoreUnusedComment = ({
+    filePath,
+    sourceFileText,
+    declarationNode,
+    referencesLength,
+    declarationName
+}: DeclarationInfo) => {
+    for (const range of declarationNode.getLeadingCommentRanges()) {
+        if (
+            sourceFileText
+                .slice(range.getPos(), range.getEnd())
+                .includes(ignoreUnusedComment)
+        ) {
+            if (referencesLength > 1) {
+                console.log(
+                    `@ignore-unused wasn't needed for ${declarationName} - ${filePath}\n`
+                )
+            }
+            return true
+        }
+    }
+    return false
+}
 
+const indexFiles = sourceFiles.filter(
```

Comment:
Unfortunately, there are tons of `index.ts` files in the repo, but the only ones that are part of the public API are `@re-/type/src/index.ts` and `@re-/assert/src/index.ts`. I'm fine just maintaining this as a constant list of entry point files, since the only time we will be adding new ones is if we add new packages or subexports, neither of which should happen often.